lib/maths/CBayesianOptimisation.cc

```diff
@@ -372,6 +398,268 @@ double CBayesianOptimisation::kernel(const TVector& a, const TVector& x, const T
                                          (x - y));
 }
 
+std::function<bool(core::CStateRestoreTraverser&)>
+CBayesianOptimisation::restoreVector(const std::string& tag,
+                                     CBayesianOptimisation::TDoubleVec& vector) {
+    auto restorationFunction = [&tag, &vector](core::CStateRestoreTraverser& traverser) {
+        double newValue{0.0};
+        CBayesianOptimisation::TDoubleVec newVector;
+        do {
+            if (traverser.name() == tag) {
+                if (core::CStringUtils::stringToType(traverser.value(), newValue) == false) {
+                    LOG_ERROR(<< "Error restoring vector: " << traverser.value());
+                    return false;
+                }
+                newVector.emplace_back(newValue);
+            }
+        } while (traverser.next());
+        vector = std::move(newVector);
+        return true;
+    };
+    return restorationFunction;
+}
+
+std::function<bool(core::CStateRestoreTraverser&)>
+CBayesianOptimisation::restoreVector(const std::string& tag,
+                                     CBayesianOptimisation::TVector& vector) {
+    auto restorationFunction = [&tag, &vector](core::CStateRestoreTraverser& traverser) {
+        double newValue{0.0};
+        CBayesianOptimisation::TDoubleVec newVector;
+        do {
+            if (traverser.name() == tag) {
+                if (core::CStringUtils::stringToType(traverser.value(), newValue) == false) {
+                    LOG_ERROR(<< "Error restoring vector: " << traverser.value());
+                    return false;
+                }
+                newVector.emplace_back(newValue);
+            }
+        } while (traverser.next());
+        CBayesianOptimisation::TVector newTVector(newVector.size());
+        for (std::size_t i = 0u; i < newVector.size(); ++i) {
+            newTVector(i) = newVector[i];
+        }
+        vector = std::move(newTVector);
+        return true;
+    };
+    return restorationFunction;
+}
+
+std::function<bool(core::CStateRestoreTraverser&)>
+CBayesianOptimisation::restoreParameterValuePair(TVector& parameters, double& functionValue) {
```

Comment:
If you implement `acceptRestoreTraverser` on `maths::CDenseVector` then you can use `core::CPersistUtils::restore` for this and it should automatically find and use that function.